lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -198,6 +202,20 @@ def run!(verify_result: true, allow_protected_ns: false, prune: true)
 
     private
 
+    def cluster_custom_resource_definitions
+      @_cluster_custom_resource_definitions ||= begin
+        @sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
+          crd = KubernetesResource.build(namespace: @namespace, context: @context, logger: @logger,
+            definition: r_def, statsd_tags: @namespace_tags)
+
+          unless KubernetesDeploy.const_defined?(crd.kind)
+            KubernetesDeploy.const_set crd.kind, Class.new(KubernetesDeploy::CustomResource)
```

Comment:
Can we avoid making classes? This could cause problems when used as a library against two different clusters.

Also this functions should be moved to its own class.